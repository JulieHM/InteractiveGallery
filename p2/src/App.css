.frontpage {
  font-family: "Franklin Gothic Medium", "Arial Narrow", Arial, sans-serif;
  width: 100%;
  margin: auto;
}

.gallery {
  display: flex;
  flex-direction: column;
  flex-wrap: wrap;
  justify-content: space-around;
}

#box {
  display: flex;
  flex-direction: row;
  justify-content: space-evenly;
  align-items: center;
}

.frontpage > div {
  text-align: center;
}

a {
  text-decoration: none;
  color: black;
}

a:hover {
  color: white;
}

audio {
  outline: none;
}

.poetry {
  height: 390px;
  overflow: scroll;
}

/*Makes sure scrollbar on poetry is always visible*/
::-webkit-scrollbar {
  -webkit-appearance: none;
  width: 7px;
}

::-webkit-scrollbar-thumb {
  border-radius: 4px;
  background-color: rgba(0, 0, 0, 0.5);
  box-shadow: 0 0 1px rgba(255, 255, 255, 0.5);
}

/* BUTTONS */
#btnRound {
  border-radius: 60%;
  border-style: none;
}
button:hover {
  background-color: #ddd;
  color: black;
}
button {
  padding: 7px 20px;
  margin: 2.5vw;
  outline: none;
  border-radius: 20px;
  border-style: none;
  background-color: #f1f3f4;
}
button.button_interaction {
  font-size: 24px;
}

/* BUTTONS */

#header {
  width: 100vw;
  background-color: teal;
  color: white;
  font-size: 3vw;
  padding-top: 20vw;
  text-align: center;
}

html {
  scroll-behavior: smooth !important;
}

div.gallery {
  padding: 5px;
  margin-top: 20px;
  margin-right: 3vw;
}

#gallery-box {
  display: flex;
  flex-direction: row;
  justify-content: space-evenly;
  align-items: center;
  padding: 0vw 4.5vw 0vw 4.5vw;
}

p {
  padding: 0;
  margin-top: 0px;
}

/*SMALLER SCREENS AND PHONES*/
@media screen and (max-width: 959px) {
  @supports (display: flex) {
    .frontpage {
      flex-direction: column;
    }
    h1 {
      font-size: 60px;
      padding-top: 120px;
      padding-bottom: 120px;
    }
    #header {
      margin: 0 auto;
      height: 100vh;
      width: 100vw;
    }
    #buttons_interaction {
      display: flex;
      font-size: 12px;
      flex-direction: column;
      justify-content: center;
    }

    #box {
      display: flex;
      flex-direction: column;
      justify-content: center;
    }

    button.button_interaction {
      font-size: 12px;
    }
  }
}

/*TABLET*/
@media screen and (max-width: 2318px) and (min-width: 960px) {
  @supports (display: flex) {
    .frontpage {
      flex-direction: row;
    }
    h1 {
      font-size: 94px;
      padding-bottom: 160px;
      margin: 20px;
    }
    #header {
      height: 80%;
      width: 100vw;
      padding-top: 20vw;
      padding-bottom: 28vw;
    }
    #buttons_interaction {
      flex-direction: column;
      margin: 20px;
    }

    #gallery_button {
      font-size: 62px;
    }
    #box {
      padding-bottom: 3vw;
    }

    .gallery {
      padding-top: 5vw;
    }
  }
}
