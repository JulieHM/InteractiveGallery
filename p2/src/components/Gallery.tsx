import React, { Component } from "react";
import ArtDisplay from "./ArtDisplay";
import Poetry from "./Poetry";
import Audio from "./Audio";

export default class Gallery extends Component {
  state = {
    count: Number(sessionStorage.getItem("count") || 1),
    music: true,
    author: sessionStorage.getItem("author") || "johnson"
  };

  /**
   * Called everytime the user presses the arrow to the right.
   * We do not want the number to be greater than 4, hence it is setted to 0 if it does.
   * sessionStorage lets us retreive the current information when the page is reloaded.
   *
   * Since React uses time to update count's state, it was necessary to intermediate store the new state
   * in a different variable, which is then used to set the state.
   */
  handleNext = () => {
    if (this.state.count === 4) {
      this.setState({ count: 0 });
      sessionStorage.setItem("count", JSON.stringify(0));
    } else {
      const new_count = this.state.count + 1;
      this.setState({ count: new_count });
      sessionStorage.setItem("count", JSON.stringify(this.state.count + 1));
    }
  };

  /**
   * Similiar to handleNext, except that we do not want count's state to be lower than 0, hence it is set
   * to 4 everytime it does.
   */

  handlePrevious = () => {
    if (this.state.count === 0) {
      this.setState({ count: 4 });
      sessionStorage.setItem("count", JSON.stringify(4));
    } else {
      const new_count = this.state.count - 1;
      this.setState({ count: this.state.count - 1 });
      sessionStorage.setItem("count", JSON.stringify(new_count));
    }
  };

  /**
   * The music toggles between the values true and false, hence if the value is true, we
   * want it to be false, and the other way around.
   * Due to the same problem about React not updating the states fast enough,
   * it is created a local variable here as well, which is then used to check which
   * boolean value we need to store in session storage.
   */

  handleMusic = () => {
    this.setState({ music: !this.state.music});
    const new_music = !this.state.music;
    console.log("new music: " + new_music);
    sessionStorage.setItem("audio", JSON.stringify(new_music));
  };

  componentDidMount(){
    const music = sessionStorage.getItem('audio');
    if (music){
      this.setState({count: Number(sessionStorage.getItem("count") || 1),
      music: JSON.parse(music),
      author: sessionStorage.getItem("author") || "johnson" })
    }
  }

  /**
   * Called everytime the user wants to change author.
   * Sets the state of the author to the oppsite of the current author,
   * and stores the information in session storage.
   */
  handlePoetry = () => {
    if (this.state.author === "johnson") {
      this.setState({ author: "bronte" });
      const new_author = "bronte";
      sessionStorage.setItem("author", new_author);
    } else {
      this.setState({ author: "johnson" });
      const new_author = "johnson";
      sessionStorage.setItem("author", new_author);
    }
  };


  /**
   * handleFavorite the user save the information of the currently displayed art, music and poetry,
   * which it then can display later using the handleRetrieveFavorite. Uses local storage to save the props'
   * current states into keys by using the function setItem.
   */
  handleFavorite = () => {
    const { count, author, music} = this.state;
    localStorage.setItem("count", count.toString());
    localStorage.setItem("music", music.toString());
    localStorage.setItem("author", author);

  };

  handleRetrieveFavorite = () => {
    const countString = localStorage.getItem("count");
    const author = localStorage.getItem("author");
    const musicString = localStorage.getItem("music");
    var count;
    var music;

    /**
     * Since JSON does not let us parse a value that is possible null, we check that the value is not,
     * before we parse it back to a JavaScript object.
     */
    if (countString != null && musicString != null) {
      count = JSON.parse(countString);
      music = JSON.parse(musicString);
      this.setState({ count, author, music });
    }
  };


  render() {
    console.log("current count state: " + this.state.count);
    console.log("current music state: " + this.state.music);

    return (
      <div>
        <div id="buttons_interaction">
          <button className="button_interaction" onClick={this.handleMusic}>
            Change music
          </button>
          <button className="button_interaction" onClick={this.handlePoetry}>
            Change author
          </button>
          <button className="button_interaction" onClick={this.handleFavorite}>
            Save favorite
          </button>
          <button
            className="button_interaction"
            onClick={this.handleRetrieveFavorite}
          >
            Retrieve favorite
          </button>
        </div>
        <div className="gallery">
          <Audio music={this.state.music}></Audio>
          <div id="gallery-box">
            <button
              id="btnRound"
              className="button_interaction"
              onClick={this.handlePrevious}
            >
              &#8249;
            </button>
            <button
              id="btnRound"
              className="button_interaction"
              onClick={this.handleNext}
            >
              &#8250;
            </button>
          </div>

          <div id="gallery-box">
            <div id="box">
              <ArtDisplay count={this.state.count}></ArtDisplay>
              <Poetry
                count={this.state.count}
                author={this.state.author}
              ></Poetry>
            </div>
          </div>
        </div>
      </div>
    );
  }
}
