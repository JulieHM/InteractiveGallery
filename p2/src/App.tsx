import React from "react";
import "./App.css";
import Frontpage from "./components/Frontpage";

function App() {
  return (
    <div className="wrapper">
      <Frontpage />
    </div>
  );
}

export default App;
