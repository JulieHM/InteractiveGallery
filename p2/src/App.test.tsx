import React from 'react';
import App from './App';
import Gallery from './components/Gallery';
import renderer from 'react-test-renderer';


it('renders correctly', () => {
  const tree = renderer.create(<App/>).toJSON()
  expect(tree).toMatchSnapshot()
})


