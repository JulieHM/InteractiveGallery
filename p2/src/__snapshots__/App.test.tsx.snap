// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="wrapper"
>
  <div
    className="frontpage"
  >
    <div
      id="header"
    >
      <h1>
        Your future interactive gallery
      </h1>
      <a
        href="#gallery"
        id="gallery_button"
      >
        View gallery
      </a>
    </div>
    <div
      id="gallery"
    >
      <div>
        <div
          id="buttons_interaction"
        >
          <button
            className="button_interaction"
            onClick={[Function]}
          >
            Change music
          </button>
          <button
            className="button_interaction"
            onClick={[Function]}
          >
            Change author
          </button>
          <button
            className="button_interaction"
            onClick={[Function]}
          >
            Save favorite
          </button>
          <button
            className="button_interaction"
            onClick={[Function]}
          >
            Retrieve favorite
          </button>
        </div>
        <div
          className="gallery"
        >
          <div
            id="audioWrapper"
          >
            <audio
              controls={true}
            >
              <source
                src="bensound-buddy.mp3"
                type="audio/mpeg"
              />
            </audio>
          </div>
          <div
            id="gallery-box"
          >
            <button
              className="button_interaction"
              id="btnRound"
              onClick={[Function]}
            >
              ‹
            </button>
            <button
              className="button_interaction"
              id="btnRound"
              onClick={[Function]}
            >
              ›
            </button>
          </div>
          <div
            id="gallery-box"
          >
            <div
              id="box"
            >
              <div
                className="gallery"
              >
                <div
                  className="artworks"
                >
                  <svg
                    height="300"
                    style={
                      Object {
                        "backgroundColor": "pink",
                      }
                    }
                    version="1.1"
                    viewBox="0 0 610 600"
                    width="300"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <clipPath
                      id="_face-clip"
                    >
                      <path
                        d="M297.171,295.773l0,-0.533c0,-3.485 2.829,-6.315 6.315,-6.315l3.028,0c3.486,0 6.315,2.83 6.315,6.315l0,0.533c1.056,0.148 1.879,1.309 1.879,2.717c0,1.409 -0.823,2.57 -1.879,2.718l0,3.552c0,3.485 -2.829,6.315 -6.315,6.315l-3.028,0c-3.486,0 -6.315,-2.83 -6.315,-6.315l0,-3.552c-1.056,-0.148 -1.879,-1.309 -1.879,-2.718c0,-1.408 0.823,-2.569 1.879,-2.717Z"
                      />
                    </clipPath>
                    <g
                      clipPath="url(#_face-clip)"
                    >
                      <circle
                        cx="237.272"
                        cy="129.753"
                        fill="#fff"
                        id="sun-second"
                        r="30.27"
                        style={
                          Object {
                            "fill": "orange",
                          }
                        }
                      />
                      <g
                        id="tree-shadow-second"
                      >
                        <path
                          className="hive"
                          d="M138.542,345.677l83.918,0l0,-129.208c0,0 -3.719,-27.145 -26.964,-27.144c-11.406,0.001 -17.799,10.437 -21.55,19.465c-3.751,9.029 4.505,26.409 -0.955,34.708c-7.021,10.668 -11.427,16.387 -13.504,25.284c-2.077,8.898 3.724,19.245 1.041,28.098c-2.769,9.134 -13.99,18.573 -17.655,26.705c-3.02,6.704 -4.331,22.092 -4.331,22.092Z"
                          fill="#2d845c"
                          id="tree2"
                          style={
                            Object {
                              "fill": "orange",
                            }
                          }
                        />
                        <path
                          d="M194.848,345.677l27.612,0l0,-129.208l-27.612,129.208Z"
                          fill="#262e33"
                          id="shadow2"
                        />
                      </g>
                      <g
                        id="house-second"
                      >
                        <path
                          d="M377.005,162.246c0,-1.473 -1.196,-2.668 -2.668,-2.668l-17.92,0c-1.472,0 -2.668,1.195 -2.668,2.668l0,13.076l23.256,0l0,-13.076Z"
                          fill="#262e33"
                          id="chimney2"
                        />
                        <rect
                          fill="#fff"
                          height="42"
                          id="window2"
                          width="42.7"
                          x="283.6"
                          y="279"
                        />
                        <path
                          d="M339.398,271.958l-116.938,0l0,73.719l116.938,0l0,-73.719Zm-20.047,15.766c0,-1.015 -0.824,-1.839 -1.839,-1.839l-25.024,0c-1.015,0 -1.839,0.824 -1.839,1.839l0,24.552c0,1.015 0.824,1.839 1.839,1.839l25.024,0c1.015,0 1.839,-0.824 1.839,-1.839l0,-24.552Z"
                          fill="#ffd7c4"
                          id="wall-beige2"
                        />
                        <path
                          d="M244.085,316.98c0.05,-8.047 6.462,-14.564 14.352,-14.564c7.92,0 14.351,6.569 14.351,14.659l0,26.602l-28.703,0l0,-26.602l0,-0.095Z"
                          fill="#fff"
                          id="door2"
                        />
                        <path
                          d="M377.005,175.322l-116.938,0l-37.607,40.174l116.938,0l37.607,-40.174Z"
                          fill="#ffdd47"
                          id="roof2"
                        />
                        <path
                          d="M409.56,219.094l-32.555,-43.772l-37.607,40.174l0,130.181l70.162,0l0,-126.583Z"
                          fill="#fff"
                          id="side2"
                        />
                        <rect
                          fill="#ffc524"
                          height="57.547"
                          id="wall-yellow2"
                          width="116.938"
                          x="222.46"
                          y="215.496"
                        />
                      </g>
                      <rect
                        fill="#ffd7c4"
                        height="84.808"
                        id="beach-second"
                        width="610"
                        x="0"
                        y="344.677"
                      />
                      <g
                        id="yacht-second"
                      >
                        <path
                          d="M443.063,296.904c0.301,0.136 0.495,0.436 0.495,0.766c0,0.33 -0.194,0.63 -0.495,0.766c-5.056,2.275 -19.639,8.837 -19.639,8.837l0,-19.206c0,0 14.583,6.562 19.639,8.837Z"
                          fill="#ffdd47"
                          id="flag2"
                        />
                        <path
                          d="M351.991,398.4c-0.222,0.337 -0.242,0.77 -0.05,1.125c0.191,0.356 0.563,0.578 0.966,0.578c11.565,0 68.383,0 68.383,0l0,-106.701c0,0 -60.279,91.333 -69.299,104.998Z"
                          fill="#2d845c"
                          id="sail-left2"
                          style={
                            Object {
                              "fill": "orange",
                            }
                          }
                        />
                        <path
                          d="M493.136,397.894c0.327,0.407 0.392,0.966 0.167,1.437c-0.226,0.472 -0.702,0.772 -1.225,0.772c-13.154,0 -67.587,0 -67.587,0l0,-87.495c0,0 57.706,71.695 68.645,85.286Z"
                          fill="#56b49f"
                          id="sail-right2"
                          style={
                            Object {
                              "fill": "orange",
                            }
                          }
                        />
                        <path
                          d="M424.491,288.601c0,-0.884 -0.717,-1.601 -1.6,-1.601c-0.884,0 -1.601,0.717 -1.601,1.601l0,129.641l3.201,0l0,-129.641Z"
                          fill="#262e33"
                          id="pole2"
                        />
                        <path
                          d="M382.681,450.252c-16.928,-0.962 -32.179,-18.891 -30.71,-34.144l141.268,0c1.469,15.253 -13.783,33.182 -30.711,34.144l-79.847,0Z"
                          fill="#fff"
                          id="deck2"
                        />
                      </g>
                    </g>
                    <path
                      d="M319.343,287.73c0,-1.014 -0.823,-1.838 -1.838,-1.838l-25.01,0c-1.015,0 -1.838,0.824 -1.838,1.838l0,24.54c0,1.014 0.823,1.838 1.838,1.838l25.01,0c1.015,0 1.838,-0.824 1.838,-1.838l0,-24.54Zm-22.172,8.043l0,-0.533c0,-3.485 2.829,-6.315 6.315,-6.315l3.028,0c3.486,0 6.315,2.83 6.315,6.315l0,0.533c1.056,0.148 1.879,1.309 1.879,2.717c0,1.409 -0.823,2.57 -1.879,2.718l0,3.552c0,3.485 -2.829,6.315 -6.315,6.315l-3.028,0c-3.486,0 -6.315,-2.83 -6.315,-6.315l0,-3.552c-1.056,-0.148 -1.879,-1.309 -1.879,-2.718c0,-1.408 0.823,-2.569 1.879,-2.717Z"
                      fill="#f4efe0"
                      id="frame-face-small"
                    />
                    <g
                      id="zoom"
                    >
                      <g
                        id="tree-shadow"
                      >
                        <path
                          d="M138.542,345.677l83.918,0l0,-129.208c0,0 -3.719,-27.145 -26.964,-27.144c-11.406,0.001 -17.799,10.437 -21.55,19.465c-3.751,9.029 4.505,26.409 -0.955,34.708c-7.021,10.668 -11.427,16.387 -13.504,25.284c-2.077,8.898 3.724,19.245 1.041,28.098c-2.769,9.134 -13.99,18.573 -17.655,26.705c-3.02,6.704 -4.331,22.092 -4.331,22.092Z"
                          fill="#2d845c"
                          id="tree"
                          style={
                            Object {
                              "fill": "orange",
                            }
                          }
                        />
                        <path
                          d="M194.848,345.677l27.612,0l0,-129.208l-27.612,129.208Z"
                          fill="#262e33"
                          id="shadow"
                        />
                      </g>
                      <g
                        id="house"
                      >
                        <path
                          d="M377.005,162.246c0,-1.473 -1.196,-2.668 -2.668,-2.668l-17.92,0c-1.472,0 -2.668,1.195 -2.668,2.668l0,13.076l23.256,0l0,-13.076Z"
                          fill="#262e33"
                          id="chimney"
                        />
                        <rect
                          fill="#fff"
                          height="42"
                          id="window"
                          width="42.7"
                          x="283.6"
                          y="279"
                        />
                        <path
                          d="M339.398,271.958l-116.938,0l0,73.719l116.938,0l0,-73.719Zm-20.047,15.766c0,-1.015 -0.824,-1.839 -1.839,-1.839l-25.024,0c-1.015,0 -1.839,0.824 -1.839,1.839l0,24.552c0,1.015 0.824,1.839 1.839,1.839l25.024,0c1.015,0 1.839,-0.824 1.839,-1.839l0,-24.552Z"
                          fill="#ffd7c4"
                          id="wall-beige"
                        />
                        <path
                          d="M244.085,316.98c0.05,-8.047 6.462,-14.564 14.352,-14.564c7.92,0 14.351,6.569 14.351,14.659l0,26.602l-28.703,0l0,-26.602l0,-0.095Z"
                          fill="#fff"
                          id="door"
                        />
                        <path
                          d="M377.005,175.322l-116.938,0l-37.607,40.174l116.938,0l37.607,-40.174Z"
                          fill="#ffdd47"
                          id="roof"
                        />
                        <path
                          d="M409.56,219.094l-32.555,-43.772l-37.607,40.174l0,130.181l70.162,0l0,-126.583Z"
                          fill="#fff"
                          id="side"
                        />
                        <rect
                          fill="#ffc524"
                          height="57.547"
                          id="wall-yellow"
                          width="116.938"
                          x="222.46"
                          y="215.496"
                        />
                      </g>
                      <rect
                        fill="#ffd7c4"
                        height="84.808"
                        id="beach"
                        width="610"
                        x="0"
                        y="344.677"
                      />
                      <rect
                        fill="#56b49f"
                        height="170"
                        id="sea"
                        width="550"
                        x="30"
                        y="429.56"
                      />
                      <circle
                        cx="237.272"
                        cy="129.753"
                        fill="#fff"
                        id="sun"
                        r="30.27"
                        style={
                          Object {
                            "fill": "orange",
                          }
                        }
                      />
                      <g
                        id="yacht"
                      >
                        <path
                          d="M443.063,296.904c0.301,0.136 0.495,0.436 0.495,0.766c0,0.33 -0.194,0.63 -0.495,0.766c-5.056,2.275 -19.639,8.837 -19.639,8.837l0,-19.206c0,0 14.583,6.562 19.639,8.837Z"
                          fill="#ffdd47"
                          id="flag"
                        />
                        <path
                          d="M351.991,398.4c-0.222,0.337 -0.242,0.77 -0.05,1.125c0.191,0.356 0.563,0.578 0.966,0.578c11.565,0 68.383,0 68.383,0l0,-106.701c0,0 -60.279,91.333 -69.299,104.998Z"
                          fill="#2d845c"
                          id="sail-left"
                          style={
                            Object {
                              "fill": "orange",
                            }
                          }
                        />
                        <path
                          d="M493.136,397.894c0.327,0.407 0.392,0.966 0.167,1.437c-0.226,0.472 -0.702,0.772 -1.225,0.772c-13.154,0 -67.587,0 -67.587,0l0,-87.495c0,0 57.706,71.695 68.645,85.286Z"
                          fill="#56b49f"
                          id="sail-right"
                          style={
                            Object {
                              "fill": "orange",
                            }
                          }
                        />
                        <path
                          d="M424.491,288.601c0,-0.884 -0.717,-1.601 -1.6,-1.601c-0.884,0 -1.601,0.717 -1.601,1.601l0,129.641l3.201,0l0,-129.641Z"
                          fill="#262e33"
                          id="pole"
                        />
                        <path
                          d="M382.681,450.252c-16.928,-0.962 -32.179,-18.891 -30.71,-34.144l141.268,0c1.469,15.253 -13.783,33.182 -30.711,34.144l-79.847,0Z"
                          fill="#fff"
                          id="deck"
                        />
                      </g>
                    </g>
                    <path
                      d="M610,39.086c0,-21.572 -17.514,-39.086 -39.086,-39.086l-531.828,0c-21.572,0 -39.086,17.514 -39.086,39.086l0,521.828c0,21.572 17.514,39.086 39.086,39.086l531.828,0c21.572,0 39.086,-17.514 39.086,-39.086l0,-521.828Zm-471.485,171.03l0,-11.327c0,-74.116 60.173,-134.289 134.289,-134.289l64.392,0c74.116,0 134.289,60.173 134.289,134.289l0,11.327c22.452,3.138 39.956,27.829 39.956,57.782c0,29.952 -17.504,54.643 -39.956,57.781l0,75.532c0,74.116 -60.173,134.289 -134.289,134.289l-64.392,0c-74.116,0 -134.289,-60.173 -134.289,-134.289l0,-75.532c-22.452,-3.138 -39.956,-27.829 -39.956,-57.781c0,-29.953 17.504,-54.644 39.956,-57.782Z"
                      fill="#f4efe0"
                      id="frame-face"
                    />
                  </svg>
                </div>
                <button
                  onClick={[Function]}
                >
                  Change color theme
                </button>
              </div>
              <div
                className="poetry"
              >
                <p>
                  Author: 
                  Ben Johnson
                </p>
                <br />
                <p>
                  
                </p>
                <br />
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
